integration-tests/src/test/java/com/datastax/oss/driver/api/core/ProtocolVersionMixedClusterIT.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2017-2017 DataStax Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core;
+
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.testinfra.utils.ConditionChecker;
+import com.datastax.oss.driver.internal.core.context.InternalDriverContext;
+import com.datastax.oss.driver.internal.testinfra.cluster.TestConfigLoader;
+import com.datastax.oss.simulacron.common.cluster.ClusterSpec;
+import com.datastax.oss.simulacron.common.cluster.DataCenterSpec;
+import com.datastax.oss.simulacron.common.cluster.QueryLog;
+import com.datastax.oss.simulacron.server.AddressResolver;
+import com.datastax.oss.simulacron.server.BoundCluster;
+import com.datastax.oss.simulacron.server.BoundNode;
+import com.datastax.oss.simulacron.server.Server;
+import java.net.InetSocketAddress;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static com.datastax.oss.driver.assertions.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+/**
+ * Covers protocol re-negotiation with a mixed cluster: if, after the initial connection and the
+ * first node list refresh, we find out that some nodes only support a lower version, reconnect the
+ * control connection immediately.
+ */
+public class ProtocolVersionMixedClusterIT {
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  private static Server server;
+
+  @BeforeClass
+  public static void setup() {
+    server = Server.builder().withAddressResolver(new AddressResolver.Inet4Resolver(9043)).build();
+  }
+
+  @AfterClass
+  public static void teardown() {
+    server.close();
+  }
+
+  @Test
+  public void should_downgrade_if_peer_does_not_support_negotiated_version() {
```

Comment:
It's peculiar that `should_downgrade_if_peer_does_not_support_negotiated_version` fails if run after `should_not_downgrade_and_force_down_old_nodes_if_version_forced`  I don't see any shared state between tests other than simulacron itself, will look into it.

Do you mind if I update these tests?  I have a change pending on simulacron to expose the incoming `Frame` in `QueryLog` (that way we can interact with that if we need more info, including protocol version), and I will also seek to consolidate the simulacron use between tests.